packages/react-dom/src/client/ReactDOM.js

```diff
@@ -160,9 +159,11 @@ setRestoreImplementation(restoreControlledState);
 export type DOMContainer =
   | (Element & {
       _reactRootContainer: ?Root,
+      _reactHasBeenPassedToCreateRootDEV: ?boolean,
```

Comment:
Shouldn't need this because `_reactRootContainer` only exists on legacy roots.

Reply:
My primary goal was to detect that a container previously passed to `createRoot` is *then* passed to `unmountComponentAtNode`. I don't think I can do this without some kind of a marker on the *new* container. I can't check its first child because `createRoot` doesn't clear children (and so could mount at arbitrary point). Scanning all children seems not great.

Reply:
```js
const reactHasBeenPassedToCreateRoot = domContainer._reactRootContainer === null;
```

Reply:
But isn't that used by `ReactDOM.render` too? Since it's expressed via `createRoot` internally.

Reply:
`ReactDOM.render` uses the `ReactRoot` type, but it doesn't call `createRoot` directly.

The only place `_reactRootContainer` is set is in `legacyRenderSubtreeIntoContainer`. New roots don't need the expando because the root is retained in user space.

https://github.com/facebook/react/blob/b66e6e41e606745b30238689fa1a98e874b29e6e/packages/react-dom/src/client/ReactDOM.js#L556

Reply:
I think we're talking about two different cases here.

1. Detecting DOM container that's previously been used by `ReactDOM.render`. This is easy. That's what `_reactRootContainer` gives us.

2. Detecting DOM container that's previously been used by `ReactDOM.createRoot`. I don't see a way to do this currently. As you said, "new roots don't need the expando". That's why I couldn't figure out how to detect them, and added a new DEV field.

I need both checks. Not just one way around.

Reply:
> I suppose you're trying to distinguish between an empty DOM container that was never rendered by anything, and a DOM container that was rendered by createRoot, so you can provide a more helpful message.

Yep 👍